# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: unknown
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/564_exhaustive_number.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/ron/tests/564_exhaustive_number.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -0,0 +1,40 @@
+use ron::Number;
+
+#[test]
+fn exhaustive_number_match() {
+    match Number::U8(42) {
+        Number::I8(v) => println!("{v}"),
+        Number::I16(v) => println!("{v}"),
+        Number::I32(v) => println!("{v}"),
+        Number::I64(v) => println!("{v}"),
+        #[cfg(feature = "integer128")]
+        Number::I128(v) => println!("{v}"),
+        Number::U8(v) => println!("{v}"),
+        Number::U16(v) => println!("{v}"),
+        Number::U32(v) => println!("{v}"),
+        Number::U64(v) => println!("{v}"),
+        #[cfg(feature = "integer128")]
+        Number::U128(v) => println!("{v}"),
+        Number::F32(v) => println!("{}", v.0),
+        Number::F64(v) => println!("{}", v.0),
+        #[cfg(not(doc))]
+        Number::__NonExhaustive(never) => never.never(),
+    }
+}
+
+#[test]
+fn non_exhaustive_number_match() {
+    match Number::U8(42) {
+        Number::I8(v) => println!("{v}"),
+        Number::I16(v) => println!("{v}"),
+        Number::I32(v) => println!("{v}"),
+        Number::I64(v) => println!("{v}"),
+        Number::U8(v) => println!("{v}"),
+        Number::U16(v) => println!("{v}"),
+        Number::U32(v) => println!("{v}"),
+        Number::U64(v) => println!("{v}"),
+        Number::F32(v) => println!("{}", v.0),
+        Number::F64(v) => println!("{}", v.0),
+        v => println!("{v:?}"),
+    }
+}